tests/ui/transmute_ptr_to_ptr.stderr

```diff
@@ -1,3 +1,17 @@
+error: transmute from a type (`&T`) to itself
+  --> $DIR/transmute_ptr_to_ptr.rs:8:5
+   |
+LL |     std::mem::transmute::<&'a T, &'static T>(t)
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::useless-transmute` implied by `-D warnings`
```

Comment:
We explicitly **don't** want to lint this. See my comment in https://github.com/rust-lang/rust-clippy/pull/5333#discussion_r394277171